File: app/src/main/java/ru/p3tr0vich/fuel/views/FuelingTotalView.java
package ru.p3tr0vich.fuel.views;

import android.support.annotation.Nullable;
import android.support.annotation.Size;

import java.util.List;

import ru.p3tr0vich.fuel.models.FuelingRecord;

public interface FuelingTotalView {
    void setAverage(float average);

    void setCostSum(float costSum);

    void setLastConsumption(float lastConsumption);

    void setEstimatedMileage(float estimatedMileage);

    void setEstimatedTotal(float estimatedTotal);

    void onFuelingRecordsChanged(@Nullable List<FuelingRecord> fuelingRecords);

    void onLastFuelingRecordsChanged(@Nullable List<FuelingRecord> fuelingRecords);

    void destroy();
}

File: app/src/main/kotlin/views/FuelingTotalView.kt
package ru.p3tr0vich.fuel.views

import ru.p3tr0vich.fuel.models.FuelingRecord

interface FuelingTotalView {
    fun setAverage(value: Float)

    fun setCostSum(value: Float)

    fun setLastConsumption(value: Float)

    fun setEstimatedMileage(value: Float)

    fun setEstimatedTotal(value: Float)

    fun onFuelingRecordsChanged(fuelingRecords: List<FuelingRecord>?)

    fun onLastFuelingRecordsChanged(fuelingRecords: List<FuelingRecord>?)

    fun destroy()
}